### ISSUE TITLE
http calls are taking too long to fetch data

### ISSUE THREAD

**Issue by aditya234** [flutter/flutter#40085]
I compared the response time of http calls in flutter with other frameworks like angular and even Django. I am able to deduce that in Flutter http calls are taking almost twice the normal time to resolve and get a response.
Can you please guild me through a relatively optimized method of making http requests?
I am currently using http package to make these calls

**Comment by Omaryasirm** [comment]
I'm also noticing a big delay with api calls. On postman it takes 500-600ms to get response but with flutter it take up to 2 seconds. Not sure if this is normal or if I'm doing something wrong but this is big problem for me as I have to call 4-5 apis in a row one after the other and if each one takes 2 seconds then I'm waiting 10 seconds for my page to load which leads to bad user experience.

**Comment by khawarkhan** [comment]
@Omaryasirm, @aditya234, did you guys find the solution? I'm facing same problem

**Comment by aditya234** [comment]
Not until now!

**Comment by aditya234** [comment]
<img width="1171" alt="Screenshot 2020-09-01 at 12 48 53 PM" src="https://user-images.githubusercontent.com/20707982/91814278-810f6f80-ec51-11ea-8a4f-7fe7dcb49f91.png">
`adityasoni@Adityas-MacBook-Pro  ~  flutter run --verbose                                                                          ✔  2712  12:48:21
[  +11 ms] executing: [/Users/adityasoni/development/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +38 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] f7a6a7906be96d2288f5d63a5a54c515a6e987fe
[        ] executing: [/Users/adityasoni/development/flutter/] git tag --contains HEAD
[ +177 ms] Exit code 0 from: git tag --contains HEAD
[   +2 ms] 1.17.1
[   +8 ms] executing: [/Users/adityasoni/development/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +9 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/stable
[        ] executing: [/Users/adityasoni/development/flutter/] git ls-remote --get-url origin
[  +10 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +93 ms] executing: [/Users/adityasoni/development/flutter/] git rev-parse --abbrev-ref HEAD
[  +15 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[   +8 ms] executing: sw_vers -productName
[  +25 ms] Exit code 0 from: sw_vers -productName
[        ] Mac OS X
[        ] executing: sw_vers -productVersion
[  +19 ms] Exit code 0 from: sw_vers -productVersion
[        ] 10.15.3
[        ] executing: sw_vers -buildVersion
[  +14 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 19D76
[  +28 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +23 ms] executing: /Users/adityasoni/Library/Android/sdk/platform-tools/adb devices -l
[   +9 ms] executing: /usr/bin/xcode-select --print-path
[   +7 ms] Exit code 0 from: /usr/bin/xcode-select --print-path
[        ] /Applications/Xcode.app/Contents/Developer
[   +1 ms] executing: /usr/bin/xcodebuild -version
[ +925 ms] Exit code 0 from: /usr/bin/xcodebuild -version
[        ] Xcode 11.3.1
           Build version 11C504
[   +6 ms] executing: xcrun --find xcdevice
[  +11 ms] Exit code 0 from: xcrun --find xcdevice
[        ] /Applications/Xcode.app/Contents/Developer/usr/bin/xcdevice
[        ] executing: xcrun xcdevice list --timeout 2
[   +3 ms] /usr/bin/xcrun simctl list --json devices
[        ] executing: /usr/bin/xcrun simctl list --json devices
[  +76 ms] List of devices attached
[ +255 ms] {
                      "devices" : {
                        "com.apple.CoreSimulator.SimRuntime.watchOS-6-1" : [
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple Watch Series 4 - 40mm",
                            "udid" : "16975DCC-0F3E-4863-82F5-DA2D5E6C9EF9"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple Watch Series 4 - 44mm",
                            "udid" : "05202ABD-E2A8-44EC-961B-15C137984CF1"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple Watch Series 5 - 40mm",
                            "udid" : "56FE81EC-6AB2-4A3E-8D5F-252CAE888895"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple Watch Series 5 - 44mm",
                            "udid" : "247525CD-F4A6-4533-9939-8E8174143D5E"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.tvOS-13-3" : [
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple TV",
                            "udid" : "07864BD4-BF40-4514-81B5-9D23EFAA2789"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple TV 4K",
                            "udid" : "365E9B60-6005-4667-9A40-58DA87970227"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "Apple TV 4K (at 1080p)",
                            "udid" : "E1B8A5E7-2060-4A90-99B2-6098592ACF93"
                          }
                        ],
                        "com.apple.CoreSimulator.SimRuntime.iOS-13-3" : [
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone SE",
                            "udid" : "D9665878-92D7-4D45-B8DE-111162662405"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone 8",
                            "udid" : "6D6BD131-3362-44DB-8C82-BB15F68B918E"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone 8 Plus",
                            "udid" : "AB7DE5BD-006E-430B-B455-798867094DF1"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone 11",
                            "udid" : "78C1749B-CC0F-434A-9EE1-FD2CD091D807"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone 11 Pro",
                            "udid" : "C55D3149-7B32-498F-96F8-5778BC628BB4"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPhone 11 Pro Max",
                            "udid" : "46D9CF6A-6375-42EB-9EFB-82B6523D9BD3"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPad Pro (9.7-inch)",
                            "udid" : "D725447A-CCF2-4279-9FCB-D48BE2FDF893"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPad (7th generation)",
                            "udid" : "2826E6FE-79B2-43D3-8A62-7C752E945319"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPad Pro (11-inch)",
                            "udid" : "BF3B8DD4-46E5-48C3-A64C-64D9103E63CC"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPad Pro (12.9-inch) (3rd generation)",
                            "udid" : "EFCDDC22-4F2B-487F-B5CD-7BDCFC5CD4D3"
                          },
                          {
                            "state" : "Shutdown",
                            "isAvailable" : true,
                            "name" : "iPad Air (3rd generation)",
                            "udid" : "758D373F-EF12-43F1-BCCD-655697CB89E2"
                          }
                        ]
                      }
                    }
[+3077 ms] [
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "6.1.1 (17S445)",
                        "available" : true,
                        "platform" : "com.apple.platform.watchsimulator",
                        "modelCode" : "Watch4,4",
                        "identifier" : "05202ABD-E2A8-44EC-961B-15C137984CF1",
                        "architecture" : "i386",
                        "modelName" : "Apple Watch Series 4 - 44mm",
                        "name" : "Apple Watch Series 4 - 44mm"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone8,4",
                        "identifier" : "D9665878-92D7-4D45-B8DE-111162662405",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone SE",
                        "name" : "iPhone SE"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "6.1.1 (17S445)",
                        "available" : true,
                        "platform" : "com.apple.platform.watchsimulator",
                        "modelCode" : "Watch4,3",
                        "identifier" : "16975DCC-0F3E-4863-82F5-DA2D5E6C9EF9",
                        "architecture" : "i386",
                        "modelName" : "Apple Watch Series 4 - 40mm",
                        "name" : "Apple Watch Series 4 - 40mm"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone12,1",
                        "identifier" : "78C1749B-CC0F-434A-9EE1-FD2CD091D807",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone 11",
                        "name" : "iPhone 11"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "6.1.1 (17S445)",
                        "available" : true,
                        "platform" : "com.apple.platform.watchsimulator",
                        "modelCode" : "Watch5,3",
                        "identifier" : "56FE81EC-6AB2-4A3E-8D5F-252CAE888895",
                        "architecture" : "i386",
                        "modelName" : "Apple Watch Series 5 - 40mm",
                        "name" : "Apple Watch Series 5 - 40mm"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPad11,3",
                        "identifier" : "758D373F-EF12-43F1-BCCD-655697CB89E2",
                        "architecture" : "x86_64",
                        "modelName" : "iPad Air (3rd generation)",
                        "name" : "iPad Air (3rd generation)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone12,3",
                        "identifier" : "C55D3149-7B32-498F-96F8-5778BC628BB4",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone 11 Pro",
                        "name" : "iPhone 11 Pro"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone10,4",
                        "identifier" : "6D6BD131-3362-44DB-8C82-BB15F68B918E",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone 8",
                        "name" : "iPhone 8"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "6.1.1 (17S445)",
                        "available" : true,
                        "platform" : "com.apple.platform.watchsimulator",
                        "modelCode" : "Watch5,4",
                        "identifier" : "247525CD-F4A6-4533-9939-8E8174143D5E",
                        "architecture" : "i386",
                        "modelName" : "Apple Watch Series 5 - 44mm",
                        "name" : "Apple Watch Series 5 - 44mm"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPad8,5",
                        "identifier" : "EFCDDC22-4F2B-487F-B5CD-7BDCFC5CD4D3",
                        "architecture" : "x86_64",
                        "modelName" : "iPad Pro (12.9-inch) (3rd generation)",
                        "name" : "iPad Pro (12.9-inch) (3rd generation)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17K446)",
                        "available" : true,
                        "platform" : "com.apple.platform.appletvsimulator",
                        "modelCode" : "AppleTV5,3",
                        "identifier" : "07864BD4-BF40-4514-81B5-9D23EFAA2789",
                        "architecture" : "x86_64",
                        "modelName" : "Apple TV",
                        "name" : "Apple TV"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17K446)",
                        "available" : true,
                        "platform" : "com.apple.platform.appletvsimulator",
                        "modelCode" : "AppleTV6,2",
                        "identifier" : "E1B8A5E7-2060-4A90-99B2-6098592ACF93",
                        "architecture" : "x86_64",
                        "modelName" : "Apple TV 4K (at 1080p)",
                        "name" : "Apple TV 4K (at 1080p)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17K446)",
                        "available" : true,
                        "platform" : "com.apple.platform.appletvsimulator",
                        "modelCode" : "AppleTV6,2",
                        "identifier" : "365E9B60-6005-4667-9A40-58DA87970227",
                        "architecture" : "x86_64",
                        "modelName" : "Apple TV 4K",
                        "name" : "Apple TV 4K"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPad6,4",
                        "identifier" : "D725447A-CCF2-4279-9FCB-D48BE2FDF893",
                        "architecture" : "x86_64",
                        "modelName" : "iPad Pro (9.7-inch)",
                        "name" : "iPad Pro (9.7-inch)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone10,5",
                        "identifier" : "AB7DE5BD-006E-430B-B455-798867094DF1",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone 8 Plus",
                        "name" : "iPhone 8 Plus"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPad7,12",
                        "identifier" : "2826E6FE-79B2-43D3-8A62-7C752E945319",
                        "architecture" : "x86_64",
                        "modelName" : "iPad (7th generation)",
                        "name" : "iPad (7th generation)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPad8,1",
                        "identifier" : "BF3B8DD4-46E5-48C3-A64C-64D9103E63CC",
                        "architecture" : "x86_64",
                        "modelName" : "iPad Pro (11-inch)",
                        "name" : "iPad Pro (11-inch)"
                      },
                      {
                        "simulator" : true,
                        "operatingSystemVersion" : "13.3 (17C45)",
                        "available" : true,
                        "platform" : "com.apple.platform.iphonesimulator",
                        "modelCode" : "iPhone12,5",
                        "identifier" : "46D9CF6A-6375-42EB-9EFB-82B6523D9BD3",
                        "architecture" : "x86_64",
                        "modelName" : "iPhone 11 Pro Max",
                        "name" : "iPhone 11 Pro Max"
                      }
                    ]
[   +8 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[  +16 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[  +18 ms] "flutter run" took 4,568ms.
Error: No pubspec.yaml file found.
This command should be run from the root of your Flutter project.
Do not run this command from the root of your git clone of Flutter.

#0      FlutterCommand.validateCommand (package:flutter_tools/src/runner/flutter_command.dart:803:11)
#1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:331:19)
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:703:11)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:620:33)
<asynchronous suspension>
#4      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart)
#5      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#6      _rootRun (dart:async/zone.dart:1184:13)
#7      _CustomZone.run (dart:async/zone.dart:1077:19)
#8      _runZoned (dart:async/zone.dart:1619:10)
#9      runZoned (dart:async/zone.dart:1539:10)
#10     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#11     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:610:20)
#12     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#13     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:339:21)
<asynchronous suspension>
#14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart)
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#16     _rootRun (dart:async/zone.dart:1184:13)
#17     _CustomZone.run (dart:async/zone.dart:1077:19)
#18     _runZoned (dart:async/zone.dart:1619:10)
#19     runZoned (dart:async/zone.dart:1539:10)
#20     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#21     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:288:19)
#22     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#23     new Future.sync (dart:async/future.dart:224:31)
#24     CommandRunner.run (package:args/command_runner.dart:112:14)
#25     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:231:18)
#26     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
#27     _rootRun (dart:async/zone.dart:1184:13)
#28     _CustomZone.run (dart:async/zone.dart:1077:19)
#29     _runZoned (dart:async/zone.dart:1619:10)
#30     runZonedGuarded (dart:async/zone.dart:1608:12)
#31     runZoned (dart:async/zone.dart:1536:12)
#32     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#33     run.<anonymous closure> (package:flutter_tools/runner.dart)
#34     runInContext.runnerWrapper (package:flutter_tools/src/context_runner.dart:64:18)
<asynchronous suspension>
#35     runInContext.runnerWrapper (package:flutter_tools/src/context_runner.dart)
#36     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:29)
#37     _rootRun (dart:async/zone.dart:1184:13)
#38     _CustomZone.run (dart:async/zone.dart:1077:19)
#39     _runZoned (dart:async/zone.dart:1619:10)
#40     runZoned (dart:async/zone.dart:1539:10)
#41     AppContext.run (package:flutter_tools/src/base/context.dart:149:18)
#42     runInContext (package:flutter_tools/src/context_runner.dart:67:24)
#43     run (package:flutter_tools/runner.dart:48:10)
#44     main (package:flutter_tools/executable.dart:69:9)
#45     main (file:///Users/adityasoni/development/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#46     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#47     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)



 adityasoni@Adityas-MacBook-Pro  ~ `

**Comment by pedromassangocode** [comment]
Hi @aditya234 
Please provide a minimal reproducible sample code.
Thank you

**Comment by aditya234** [comment]
If you call API [Link](https://jsonplaceholder.typicode.com/albums/1) from a rest client, it will take less time than dart. 
Instead of the above link, please also check for [link](https://jsonplaceholder.typicode.com/albums)
For getting a substantial difference in time

`import 'dart:async';
import 'dart:convert';

import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;

Future<Album> fetchAlbum() async {
  final response =
      await http.get('https://jsonplaceholder.typicode.com/albums/1');

  if (response.statusCode == 200) {
    // If the server did return a 200 OK response,
    // then parse the JSON.
    return Album.fromJson(json.decode(response.body));
  } else {
    // If the server did not return a 200 OK response,
    // then throw an exception.
    throw Exception('Failed to load album');
  }
}

class Album {
  final int userId;
  final int id;
  final String title;

  Album({this.userId, this.id, this.title});

  factory Album.fromJson(Map<String, dynamic> json) {
    return Album(
      userId: json['userId'],
      id: json['id'],
      title: json['title'],
    );
  }
}

void main() => runApp(MyApp());

class MyApp extends StatefulWidget {
  MyApp({Key key}) : super(key: key);

  @override
  _MyAppState createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  Future<Album> futureAlbum;

  @override
  void initState() {
    super.initState();
    futureAlbum = fetchAlbum();
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Fetch Data Example',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: Text('Fetch Data Example'),
        ),
        body: Center(
          child: FutureBuilder<Album>(
            future: futureAlbum,
            builder: (context, snapshot) {
              if (snapshot.hasData) {
                return Text(snapshot.data.title);
              } else if (snapshot.hasError) {
                return Text("${snapshot.error}");
              }

              // By default, show a loading spinner.
              return CircularProgressIndicator();
            },
          ),
        ),
      ),
    );
  }
}`

**Comment by aditya234** [comment]
This is a similar issue https://github.com/flutter/flutter/issues/20419

**Comment by pedromassangocode** [comment]
Hi @aditya234 
The "slow" may vary depending on the internet speed and/or the server you are using  or from where you are accessing the API.

Can you please elaborate on "slow"?
Is this only in debug or also in release mode?
Is this only in the emulator or also with a real device?
How are you measuring the speed of the requests?

I do think this may be an issue with the `package:http` package, but not closing after reading https://github.com/flutter/flutter/issues/20419#issuecomment-486799319.

Cc @zoechi @natebosch I saw your previous interaction on https://github.com/flutter/flutter/issues/20419.

**Comment by natebosch** [comment]
cc @zichangg who looked at a problem around slow DNS queries on IOs.

> I do think this may be an issue with the `package:http` package

If you can demonstrated a reproduction case where a request that is fast with `dart:io` interfaces and slow with `package:http` interfaces I'd be happy to investigate. `package:http` is very thin and has very little opportunity to make things slower than they would be otherwise.

**Comment by zichangg** [comment]
@aditya234  I strongly recommend [dartdev tool](https://flutter.dev/docs/development/tools/devtools/timeline). You are able to check the duration of http calls and sockets. You can use those stats to elaborate the speed of dart http.

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by Djihanegh** [comment]
Hello guys, im having the same issue too , i use my real device with node js as backend , other APIs work fine only this one who takes more than 300sec to load...
